### PR TITLE
[ACS-8063] Migrate editActions e2e from Protractor to Playwright

### PR DESCRIPTION
**Pull request by katarzynakita**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [X] Other... Please describe: Test migration https://hyland.atlassian.net/browse/ACS-8063

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**
Migrated edit-folder tests from Protractor to Playwright. 
Removed Protractor tests. 
Removed protractor tests github pull request workflow. 
Added Playwright edit-actions tests github pull request workflow.
Added additional wait in Viever tests as they were flaky.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: e2e/playwright/edit-actions/src/tests/edit-offline.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    await personalFiles.dataTable.selectItem(file1);
+    await personalFiles.acaHeader.clickMoreActions();
+    await personalFiles.matMenu.clickMenuItem('Edit Offline');
+    const [download] = await Promise.all([personalFiles.page.waitForEvent('download')]);
+    expect(download.suggestedFilename()).toBe(file1);
+  });
+
+  test('XAT-5305] Lock information is displayed', async ({ personalFiles }) => {
+    expect(await personalFiles.dataTable.isItemPresent(fileLocked2)).toBe(true);
+    // await expect(personalFiles.dataTable.hasLockIcon(fileLocked2)).toBe(true);
```
Comment (akashrathod28) [inline e2e/playwright/edit-actions/src/tests/edit-offline.e2e.ts:83]: can you please remove commented code. we do not keep code in comment

File: e2e/playwright/edit-actions/src/tests/edit-offline.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  });
+
+  test('[XAT-5306] Cancel Editing unlocks the file', async ({ personalFiles }) => {
+    await personalFiles.dataTable.selectItem(fileLocked);
+    await personalFiles.acaHeader.clickMoreActions();
+    await personalFiles.matMenu.clickMenuItem('Cancel Editing');
+    await personalFiles.dataTable.selectItem(fileLocked);
+
+    expect(await nodesApi.getNodeProperty(fileLockedId, 'cm:lockType'), `${fileLocked} is still locked`).not.toEqual('WRITE_LOCK');
+    // expect(await personalFiles.dataTable.hasLockIcon(fileLocked)).toBe(false);
```
Comment (akashrathod28) [inline e2e/playwright/edit-actions/src/tests/edit-offline.e2e.ts:94]: can you please remove commented code. we do not keep code in comment.

File: e2e/playwright/edit-actions/src/tests/edit-offline.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  const apiClientFactory = new ApiClientFactory();
+  let nodesApi: NodesApi;
+  let trashcanApi: TrashcanApi;
+  let fileActionsApi: FileActionsApi;
+
+  let parentPFId: string;
+  let fileLockedId: string;
+  let fileLocked2Id: string;
+
+  const username = `user-${Utils.random()}`;
```
Comment (MateuszHY) [inline e2e/playwright/edit-actions/src/tests/edit-offline.e2e.ts:38]: Out of curiosity, why we are using random ?

**Review by MateuszHY — APPROVED** [review]

File: projects/aca-playwright-shared/src/page-objects/components/dataTable/mat-menu.component.ts [hunk 4]
```diff
@@ -49,6 +48,13 @@ export class MatMenuComponent extends BaseComponent {
     await menuElement.waitFor({ state: 'detached' });
   }
 
+  async clickMenuItemFromHeaderMenu(menuItem: string): Promise<void> {
+    const menuElement = this.getMenuItemFromHeaderMenu(menuItem);
+    await menuElement.waitFor({ state: 'attached' });
+    await menuElement.click();
```
Comment (datguychen) [inline projects/aca-playwright-shared/src/page-objects/components/dataTable/mat-menu.component.ts:54]: Click action already has auto-waiting for element, so line 53 is not needed.
 
If you want to check which actions have auto-waiting in Playwright, you can read more of it here: https://playwright.dev/docs/actionability

File: projects/aca-playwright-shared/src/page-objects/components/dialogs/edit-dialog.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Page } from '@playwright/test';
+import { BaseComponent } from '../base.component';
+
+export class EditDialog extends BaseComponent {
+  private static rootElement = 'mat-dialog-container';
+
+  public editDialog = this.page.locator('mat-dialog-container');
```
Comment (datguychen) [inline projects/aca-playwright-shared/src/page-objects/components/dialogs/edit-dialog.component.ts:31]: editDialog locator is the same as rootElement. If you encounter some issues with using the rootElement, you could try this.getChild(''); to define the rootElement as a locator, but I dont think thats the best way here.

File: projects/aca-playwright-shared/src/page-objects/components/dialogs/edit-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  public titleInput = this.getChild('[data-automation-id="adf-folder-dialog-title"]');
+  public cancelButton = this.getChild('#adf-folder-cancel-button');
+  public updateButton = this.getChild('#adf-folder-create-button');
+  public descriptionInput = this.getChild('#adf-folder-description-input');
+  public nameInput = this.getChild('#adf-folder-name-input');
+  public fieldHint = this.getChild('mat-hint');
+
```
Comment (datguychen) [inline projects/aca-playwright-shared/src/page-objects/components/dialogs/edit-dialog.component.ts:38]: Please make sure that all the mat locators are listed here: https://hyland.atlassian.net/browse/ACS-8369.

If not, add/update a subtask with these locators so they won't get skipped when the devs are going to update the material locators.

**Review by DenysVuika — APPROVED** [review]

File: projects/aca-playwright-shared/src/page-objects/components/dataTable/mat-menu.component.ts (hunk 4 above)
Comment (katarzynakita) [reply projects/aca-playwright-shared/src/page-objects/components/dataTable/mat-menu.component.ts:52]: fixed

File: projects/aca-playwright-shared/src/page-objects/components/dialogs/edit-dialog.component.ts (hunk 6 above)
Comment (katarzynakita) [reply projects/aca-playwright-shared/src/page-objects/components/dialogs/edit-dialog.component.ts:38]: Both are there :)

**Pull request merged**